packaging/metadata/__init__.py

```diff
@@ -85,43 +99,27 @@ def _pkginfo_string_to_dict(
         - The result should be stored as a string-keyed dictionary.
         """
         metadata = {}  # type : Dict[str, Union[List[str], str]]
-        parsed = HeaderParser().parsestr(pkg_info_string)
+        parsed = HeaderParser().parsestr(rfc822_string)
         metadata_fields = VERSIONED_METADATA_FIELDS[parsed.get("Metadata-Version")]
 
         for key, value in parsed.items():
-
             if key in metadata_fields["MULTI"]:
-                if key not in metadata:
-                    metadata[key] = []
-                metadata[key].append(value)
+                metadata.setdefault(key, []).append(value)
             elif key in metadata_fields["TREAT_AS_MULTI"]:
                 metadata[key] = [val.strip() for val in value.split(",")]
+            elif key == "Description":
+                metadata[key] = inspect.cleandoc(value)
             else:
                 metadata[key] = value
 
+        # Handle the message payload
         payload = parsed.get_payload()
         if payload:
-
             if "Description" in metadata:
-                print("Both Description and payload given - ignoring Description")
-
+                raise Exception("Duplicate descriptions given")
```

Comment:
This should probably be an exception from `packaging.metadata.exceptions` instead.